prive/tests/test_dataset.py

```diff
@@ -100,6 +100,15 @@ def test_replace(self):
         replaced_sample = data_sample100.replace(records_in)
         self.assertEqual(data_sample100.data.shape[0] + 1, replaced_sample.data.shape[0])
 
+    def test_iter(self):
+        record_count = 0
+        for record in self.data:
+            self.assertEqual(type(record), TabularDataset)
+            self.assertEqual(record.description, self.data.description)
+            self.assertEqual(record.data.shape[0], 1)
```

Comment:
The test seems to be failing in this line with assertion error 18 != 1, maybe the iterator is not doing exactly what you expect?

Reply:
The issue is coming from the fact that `TabularDataset` is constructing a pandas `Series` object, and thus calling shape returns the length of the series (which is the number of columns).

Reply:
Fixed, ty both